Purchases/Public/RCPurchases.m

```diff
@@ -295,12 +295,18 @@ - (instancetype)initWithAppUserID:(nullable NSString *)appUserID
         };
 
         [self.identityManager configureWithAppUserID:appUserID];
-        if (!self.systemInfo.isApplicationBackgrounded) {
-            [self updateAllCachesWithCompletionBlock:callDelegate];
-        } else {
-            [self sendCachedPurchaserInfoIfAvailable];
-        }
-        
+
+        [self.systemInfo isApplicationBackgroundedWithCompletion:^(BOOL isBackgrounded) {
+            if (!isBackgrounded) {
+                dispatch_queue_t backgroundQueue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
+                dispatch_async(backgroundQueue, ^{
+                    [self updateAllCachesWithCompletionBlock:callDelegate];
+                });
+            } else {
+                [self sendCachedPurchaserInfoIfAvailable];
+            }
+        }];
```

Comment:
this is the reason for a lot of the changes in the tests - since we're calling the `updateAllCaches` method from a background queue, it will take longer to return for tests.